server/src/test/java/org/elasticsearch/rest/action/admin/indices/RestValidateQueryActionTests.java

```diff
@@ -0,0 +1,144 @@
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.admin.indices.validate.query.ValidateQueryAction;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.search.AbstractSearchTestCase;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskManager;
+import org.elasticsearch.test.rest.FakeRestChannel;
+import org.elasticsearch.test.rest.FakeRestRequest;
+import org.elasticsearch.threadpool.TestThreadPool;
+import org.elasticsearch.usage.UsageService;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.emptySet;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+public class RestValidateQueryActionTests extends AbstractSearchTestCase {
+
+    private static final Settings settings = Settings.EMPTY;
+    private static final TestThreadPool threadPool = new TestThreadPool(RestValidateQueryActionTests.class.getName());
+    private static final NodeClient client = new NodeClient(settings, threadPool);
+
+    private static final UsageService usageService = new UsageService(settings);
+    private static final RestController controller = new RestController(settings, emptySet(), null, client, null, usageService);
+    private static final RestValidateQueryAction action = new RestValidateQueryAction(settings, controller);
+
+    /**
+     * Configures {@link NodeClient} to stub {@link ValidateQueryAction} transport action.
+     * <p>
+     * This lower level of validation is out of the scope of this test.
+     */
+    @BeforeClass
+    public static void stubValidateQueryAction() {
+        final TaskManager taskManager = new TaskManager(Settings.EMPTY, threadPool, Collections.emptySet());
+
+        final TransportAction transportAction = new TransportAction(Settings.EMPTY, ValidateQueryAction.NAME,
+            new ActionFilters(Collections.emptySet()), taskManager) {
+            @Override
+            protected void doExecute(Task task, ActionRequest request, ActionListener listener) {
+            }
+        };
+
+        final Map<Action, TransportAction> actions = new HashMap<>();
+        actions.put(ValidateQueryAction.INSTANCE, transportAction);
+
+        client.initialize(actions, () -> "local", null);
+    }
+
+    @AfterClass
+    public static void terminateTestThreadPool() throws InterruptedException {
+        terminate(threadPool);
```

Comment:
can you assign all of the static variables to null as part of this method?

Reply:
Oh, yes...